### PR TITLE
Script to check connectivity to azure

### PR DESCRIPTION
**Pull request by giggsoff**
Azure use signature checking across provided headers. It would be
nice to have a tool to debug unexpected intermediate headers
modifications that may lead to errors like "The MAC signature found in
the HTTP request is not the same as any computed signature". Even
without access_key the script allow to show headers server use to
calculate the signature.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: pkg/debug/scripts/azure-check.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+string_to_sign="${request_method}\n\n\n\n\n\n\n\n\n\n\n\n${canonicalized_headers}\n${canonicalized_resource}"
+
+# Decode the Base64 encoded access key, convert to Hex.
+# shellcheck disable=SC2059
+decoded_hex_key="$(printf "$ACCESS_KEY" | base64 -d | xxd -p -c256)"
+
+# Create the HMAC signature for the Authorization header
+# shellcheck disable=SC2059
+signature=$(printf "$string_to_sign" | openssl dgst -sha256 -mac HMAC -macopt "hexkey:$decoded_hex_key" -binary | base64)
```
Comment (zededa-yuri) [inline pkg/debug/scripts/azure-check.sh:125]: might be worth to check if `openssl` returned an error

**Comment by giggsoff** [comment]
@zededa-yuri thanks for recommendations, modified.

**Review by zededa-yuri — APPROVED** [review]

**Comment by rouming** [comment]
@giggsoff @eriknordmark Let's merge this?

**Review by eriknordmark — APPROVED** [review]
Let's run the tests.

**Pull request merged**